### ISSUE TITLE
qemu support of SandyBridge

### ISSUE THREAD

**Issue by derkuci** [joyent/smartos-live#745]
It seems that SmartOS comes with qemu 0.14.1which doesn't know some of the newer cpu ids, e.g., SandyBridge.
```
# /smartdc/bin/qemu-system-x86_64 -version
QEMU emulator version 0.14.1 (qemu-kvm-devel), Copyright (c) 2003-2008 Fabrice Bellard
# /smartdc/bin/qemu-system-x86_64 -cpu SandyBridge
Unable to find x86 CPU definition
```
Is there a way for us to "upgrade" qemu in SmartOS?  Or, am I asking the right question in order to support SandyBridge instruction sets in KVM?

Thank you!

**Comment by sjorge** [comment]
You can set the CPU model to host or to qemu64,+aes,... Just add all the flags you need.

You can for example also start from core2duo and  flags.

~ sjorge

**Comment by derkuci** [comment]
I did try that ("host", "qemu64,+aes,+avx").  No avx support showed up in /proc/cpuinfo (the guest is CentOS 7), which means I can't really have SandyBridge.  The host does have avx/avx2, and the lx branded zone has no issues running code with avx instructions.

**Comment by derkuci** [comment]
This page https://wiki.smartos.org/display/DOC/Patching+the+latest+QEMU might be related, although the work there was last updated in 2012.  Seems a tough job.

**Comment by sjorge** [comment]
Have you tried with host?

~ sjorge

**Comment by derkuci** [comment]
Yes, I tried host, I tried qemu64,+avx, I tried Nehalem,+avx, etc.

**Comment by rmustacc** [comment]
The reason that AVX is masked off is that the KVM driver doesn't support saving and restoring the ymm registers through the xsave routines and propagating that to the guest. For that to be unmasked we'll need to add support for that.

**Comment by discostur** [comment]
@rmustacc are there any news on AVX for KVM on SmartOS?

**Comment by rmustacc** [comment]
On 1/28/19 10:35 , Kilian Ries wrote:
> @rmustacc are there any news on AVX for KVM on SmartOS?
> 

I suspect it's unlikely that it's going to be added, as most of the
focus is on bhyve at the moment. If someone else added it, I'd be happy
to review it though.
